sql/src/main/java/io/crate/execution/engine/window/WindowFrameState.java

```diff
@@ -43,11 +43,15 @@ int lowerBound() {
         return lowerBound;
     }
 
-    int upperBoundExclusive() {
+    public int upperBoundExclusive() {
         return upperBoundExclusive;
     }
 
     public Iterable<Object[]> getRows() {
         return rows;
     }
+
+    public Object[] lastRow() {
```

Comment:
Not sure if we wan't to generalize this already on the `WindowFrameState` It is currently specific to `last_value` - I'm afraid that by going this way we'll end up having a lot of helper methods here that are specific to 1 or few window functions.

Reply:
I think it makes sense to have the navigation over the rows be done in the WindowFrameState as it knows of boundaries and everything (and last_value will become a specific case of nth_value once we have that). If we move it this outside now, getting the last value will become a matter of iterating over all the rows to get to the last one.

Reply:
how about providing something like  :
```
    public List<Object[]> getRowsAsList() {
        return rows;
    }
```
and delegate access to the specific case as needed?